shared/settings/files/index.js

```diff
@@ -1,28 +1,110 @@
 import * as Kb from '../../common-adapters'
-import {fileUIName, isLinux} from '../../constants/platform'
+import {InlineDropdown} from '../../common-adapters/dropdown'
+import * as Platform from '../../constants/platform'
 import * as Styles from '../../styles'
 import SystemFileManagerIntegrationBanner from '../../fs/banner/system-file-manager-integration-banner/container'
 import RefreshDriverStatusOnMount from '../../fs/common/refresh-driver-status-on-mount'
+import RefreshSettings from './refresh-settings'
+import flags from '../../util/feature-flags'
 
 type Props = {|
+  areSettingsLoading: boolean,
   driverStatus: Types.DriverStatus,
-  onEnable: () => void,
   onDisable: () => void,
+  onEnable: () => void,
   onShowKextPermissionPopup: () => void,
+  spaceAvailableNotificationThreshold: number,
+  onChangedSyncNotifications: (number) => void,
+  onEnableSyncNotifications: () => void,
+  onDisableSyncNotifications: () => void,
 |}
 
+export const allowedNotificationThresholds = [
+  100 * 1024 ** 2,
+  1024 ** 3,
+  10 * 1024 ** 3,
+]
+
+export const defaultNotificationThreshold = 100 * 1024 ** 2
+
 const EnableSystemFileManagerIntegration = (props: Props) => (
   <Kb.Box style={Styles.globalStyles.flexBoxColumn}>
-    <Kb.Text type="Body">Enable Keybase in {fileUIName}</Kb.Text>
+    <Kb.Text type="Body">Enable Keybase in {Platform.fileUIName}</Kb.Text>
     <Kb.Text type="BodySmall">
       Access your Keybase files just like you normally do with your local files.
     </Kb.Text>
   </Kb.Box>
 )
 
+class ThresholdDropdownMobile extends React.PureComponent<Props, {visible: boolean}> {
+  state = {visible: false}
+  _show = () => this.setState({visible: true})
+  _hide = () => this.setState({visible: false})
+  _select = selectedIdx => this.props.onChangedSyncNotifications(selectedIdx)
+  render() {
+    return (
+      <>
+        <InlineDropdown
+          onPress={this.state.visible ? this._show : this._hide}
+          type="Body"
+          label={Constants.humanizeBytes(this.props.spaceAvailableNotificationThreshold || defaultNotificationThreshold, 0)}
+        />
+        <Kb.FloatingPicker
+          items={allowedNotificationThresholds.map(i => ({
+            label: Constants.humanizeBytes(i, 0),
+            value: i,
+          }))}
+          visible={this.state.visible}
+          selectedValue={
+            this.props.spaceAvailableNotificationThreshold || defaultNotificationThreshold
+          }
+          promptString="Pick a threshold"
+          prompt={
+            <Kb.Box2 direction="horizontal" fullWidth={true} gap="xtiny" centerChildren={true}>
+              <Kb.Text type="BodySmallSemibold">Pick a threshold</Kb.Text>
+            </Kb.Box2>
+          }
+          onCancel={this._hide}
+          onHidden={this._hide}
+          onDone={this._hide}
+          onSelect={this._select}
+        />
+      </>
+    )
+  }
+}
+
+const ThresholdDropdownDesktop = (props: Props) => (
+  <Kb.Dropdown
```

Comment:
The place the dropdown appears (not actually below the dropdown box, but centered on it) feels weird to me.

Also, the text in it is way smaller than the box itself - it feels to me like the expanded box should be the same size as the box pre-expansion. If it stays big, the text should probably be centered in it.

Finally, there should be margins around the text in the dropped-down box - right now the text is touching the borders.

![no margins](https://user-images.githubusercontent.com/4698687/58275571-f86e0800-7d5a-11e9-80b7-7f66dfde2b8f.gif)

Reply:
I was having trouble figuring out how to style this.